programs/degen-pools/src/transfers.rs

```diff
@@ -0,0 +1,49 @@
+use crate::constants::AUTHORITY_PUBKEY;
+use crate::CustomError;
+use anchor_lang::prelude::*; 
+
+#[derive(Accounts)]
+pub struct ExecuteTransfer<'info> {
+    #[account(mut)]
+    pub sender: Signer<'info>,
+    /// CHECK: This is a safe account as it is the system
+    pub system_program: AccountInfo<'info>,
+    #[account(mut, address = AUTHORITY_PUBKEY)]
+    /// CHECK: The receiver's address is validated against a known authority.
+    pub receiver: AccountInfo<'info>, 
+}
+
+pub fn handle_transfer(ctx: Context<ExecuteTransfer>, amount: u64) -> Result<()> {
+    require!(
+        amount > 0,
+        CustomError::InvalidAmount
```

Comment:
same here, if the user for some reason sends a transfer with 0 lamports, we should let it through - it should be the backend's responsibility to ignore an event that has 0 lamports